dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java

```diff
@@ -459,15 +465,16 @@ public void Test_Create_App_Descriptor_Then_Create_App_Integration_Then_Delete_A
     }
 
     @Test
-    public void Test_Protected_Hidden_Secret_And_Values_Returned_Match_Descriptor() {
+    public void Test_Secret_Serializer_Returned_Values_Match_Descriptor_Verify_Hidden_Secrets_Are_Protected() throws Exception{
 
-        final List<String> orderedParamNames = ImmutableList.of("param1","param2","param3","param4");
+        final List<String> orderedParamNames = ImmutableList.of("param1","param2","param3","param4","param5");
         //This is how the descriptor looks like.
         final SortedMap<String, ParamDescriptor> appDescriptorParamsMap = ImmutableSortedMap.of(
-                orderedParamNames.get(0), ParamDescriptor.newParam("val-1", false, Type.STRING, "label", "hint", true),
-                orderedParamNames.get(1), ParamDescriptor.newParam("true", false, Type.BOOL, "label", "hint", true),
-                orderedParamNames.get(2), ParamDescriptor.newParam("val-2", false, Type.STRING, "label", "hint", true),
-                orderedParamNames.get(3), ParamDescriptor.newParam("true", false, Type.BOOL, "label", "hint", true)
+                orderedParamNames.get(0), ParamDescriptor.newParam("", true, Type.STRING, "label", "hint", true),
```

Comment:
Is each position in the map a different test case ?

Reply:
Not really. the true purpose is validating that hidden secrets are coming back protected while the non-hidden show their original value.  I mean anything but a protected secret mask.

Reply:
I see duplicate entries, is it ok?

Reply:
should be cleaner now

Reply:
Trying to be more aligned with the rest of our library  I created an AppDescriptorDataGen